### PR TITLE
#1 Initial commit of base-maven

### PR DESCRIPTION
**Pull request by ALRubinger**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM jboss/base-jdk:8
+
+MAINTAINER Andrew Lee Rubinger (https://github.com/ALRubinger)
+
+# install wget
+USER root
+
+RUN yum install -y wget
```
Comment (rwngwn) [inline Dockerfile]: please run yum in a way of https://github.com/jboss-dockerfiles/base-jdk/blob/jdk7/Dockerfile#L8 - to preserve space

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# get maven 3.3.9
+RUN wget --no-verbose -O /tmp/apache-maven-3.3.9.tar.gz http://archive.apache.org/dist/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.tar.gz
```
Comment (rwngwn) [inline Dockerfile]: Please squash wget and other RUN instructions to one (like in https://github.com/jboss-dockerfiles/wildfly/blob/master/Dockerfile#L11) to make image smaller (in the way you did the image layers will still contain maven tarballs, etc.)

File: README.asciidoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+~~~~~~~~
+
+`docker build -t base-maven .`
+
+Example Usage
+~~~~~
+
+* `cd [workDir]`
+
+* `docker run --privileged=true -v {backtick}pwd{backtick}:/opt/jboss/myproject base-maven mvn clean install -f /opt/jboss/myproject`
```
Comment (rwngwn) [inline README.asciidoc]: why is --privileged=true required here? we should require as low permissions as possible.

**Review by rwngwn — COMMENTED** [review]
made some comments in code. thanks for submitting.

File: README.asciidoc (hunk 3 above)
Comment (ALRubinger) [reply README.asciidoc]: Needed this to be able to let the java/mvn process have access to the mounted filesystem; in this example we have the Docker container building sources contained on the host FS.  Better ideas to enable this without selinux policy violations?

File: Dockerfile (hunk 2 above)
Comment (ALRubinger) [reply Dockerfile]: +1 will do

File: Dockerfile (hunk 1 above)
Comment (ALRubinger) [reply Dockerfile]: +1 will do

File: README.asciidoc (hunk 3 above)
Comment (rwngwn) [reply README.asciidoc]: can you try to mount volume like -v {backtick}pwd{backtick}:/opt/jboss/myproject:z (append :z)? It should change context automatically.

**Comment by ALRubinger** [comment]
@l-d-j I have made new commits to address your comments; please feel free to squash into a single commit before merge to master

**Pull request merged**